### PR TITLE
Only Publish images on New point release

### PR DESCRIPTION
**Pull request by kaxil**
Github Issue: https://github.com/astronomer/issues/issues/1247

Note this does not fully address https://github.com/astronomer/issues/issues/1247

This PR only stop republishing of `astronomerinc/ap-airflow:1.10.7-buster` kind of images.
The CI will still publish images like `astronomerinc/ap-airflow:1.10.7-buster-onbuild-9373`.

Following are the things I still want to do and this PR does not solve them:

- [ ] Stop building & testing all the images when a single version is released. For example, releasing `1.10.7-10` would still build, test & scan `1.10.5-2` but won't publish it.
- [ ] If publishing new image without release, push it to astronomerio instead of astronomerinc

### CODE REVIEW

**Comment by kaxil** [comment]
Please only review last commit as the first commit depends on https://github.com/astronomer/ap-airflow/pull/77

**Comment by ashb** [comment]
Hah, by putting 

> Note this does not close astronomer/issues#1247

Github thinks its does :)

**Comment by kaxil** [comment]
> Hah, by putting
> 
> > Note this does not close [astronomer/issues#1247](https://github.com/astronomer/issues/issues/1247)
> 
> Github thinks its does :)

Haha, yes, bad choice of words

File: .circleci/config.yml [hunk 1]
```diff
@@ -715,19 +715,23 @@ commands:
...
             set -x
-            docker push "$image"
-            set +x
+            export NEW_POINT_RELEASE=true
             for tag in $(echo "<< parameters.extra_tags >>" | sed "s/,/ /g");
             do
-              set -x
               if DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect << parameters.image_name >>:$tag >/dev/null 2>&1; then
                 echo "Image with Tag (<< parameters.image_name >>:$tag) already exists"
+                export NEW_POINT_RELEASE=false
```
Comment (ashb) [inline .circleci/config.yml:723]: Shouldn't this also be in the config.yml.j2?

Comment (kaxil) [reply .circleci/config.yml:723]: Yup, missed that. I will actually create a new PR to add a  test to **test** that `config.yml` and `config.yml.j2` are in-sync

Comment (kaxil) [reply .circleci/config.yml:723]: Created: https://github.com/astronomer/ap-airflow/pull/80

**Pull request merged**